framework/src/functions/MooseParsedFunction.C

```diff
@@ -71,6 +72,24 @@ template <typename T>
 void
 MooseParsedFunctionTempl<T>::initialSetup()
 {
+  for (auto i = 0; i < (int)_vars.size(); i++)
+  {
+    // Check for non-scalar variables.
+    // First, see if the var is actually assigned to a proper scalar value
+    if (_pfb_feproblem.hasVariable(_vars[i]) && _pfb_feproblem.hasVariable(_vals[i]))
+    {
+      // Then see if the var has the same name as a function or postprocessor
+      if (!_pfb_feproblem.hasFunction(_vars[i]) &&
+          !_pfb_feproblem.hasPostprocessorValueByName(_vars[i]))
+        mooseError(
+            "The only variables supported by ParsedFunction are scalar variables, and var '" +
+            _vars[i] + "' is not scalar.");
+    }
+    if (_vars[i].find_first_of("xyzt") != std::string::npos && _vars[i].size() == 1)
+      mooseError("The variables \"x, y, z, and t\" in the ParsedFunction are pre-declared for use "
+                 "and must not be declared in \"vars\"");
+  }
+
```

Comment:
This seems ok, but since we moved this function from the base (I'm assuming this is because you don't have access to the `Problem` there), is there any object that may be missing this check (a different child of the base)?